pkg/oc/admin/diagnostics/diagnostics/client/config_contexts.go

```diff
@@ -235,26 +236,30 @@ func (d ConfigContext) Check() types.DiagnosticResult {
 
 	// Actually send a request to see if context has connectivity.
 	// Note: we cannot reuse factories as they cache the clients, so build new factory for each context.
-	projectClient, err := osclientcmd.NewFactory(kclientcmd.NewDefaultClientConfig(*d.RawConfig, &kclientcmd.ConfigOverrides{Context: *context})).OpenshiftInternalProjectClient()
-	// client create now *fails* if cannot connect to server; so, address connectivity errors below
+	temporaryFactory := osclientcmd.NewFactory(kclientcmd.NewDefaultClientConfig(*d.RawConfig, &kclientcmd.ConfigOverrides{Context: *context}))
+	clientConfig, err := temporaryFactory.ClientConfig()
 	if err == nil {
-		if projects, projerr := projectClient.Project().Projects().List(metav1.ListOptions{}); projerr != nil {
-			err = projerr
-		} else { // success!
-			list := []string{}
-			for i, project := range projects.Items {
-				if i > 9 {
-					list = append(list, "...")
-					break
+		projectClient, err := projectclientinternal.NewForConfig(clientConfig)
+		// client create now *fails* if cannot connect to server; so, address connectivity errors below
+		if err == nil {
```

Comment:
what happens when err != nil ?

Reply:
> what happens when err != nil ?

Review with whitespace ignored.  This increased nesting.  The error is handled below as it was before.